test/swift_class_test.exs

```diff
@@ -127,34 +137,77 @@ defmodule SwiftClassTest do
   end
 
   describe "class block parser" do
-    input = """
-    "color-" <> color_name do
-      foo(true)
-      color(color_name)
-      bar(false)
-    end
-
-    "color-red" do
-      color(:red)
-    end
-    """
-
-    output = [
-      {{:<>, [context: Elixir, imports: [{2, Kernel}]], ["color-", {:color_name, [], Elixir}]},
-      [
-        ["foo", [true], nil],
-        ["color", [{:color_name, [], Elixir}], nil],
-        ["bar", [false], nil]
-      ]},
-      {
-        "color-red",
-        [
-          ["color", [["IME", "red"]]]
-        ]
-      }
-
-    ]
-
-    # assert parse_class_block(input) == output
+    test "parses a simple block" do
+      input = """
+      defstyle "red-header" do
+        color(:red)
+        font(:largeTitle)
+      end
+      """
+
+      output = [
+        {
+          "red-header",
+          [
+            ["color", [["IME", "red"]], nil],
+            ["font", [["IME", "largeTitle"]], nil]
+          ]
+        }
+      ]
+
+      assert parse_class_block(input) == output
+    end
+
+    test "parses a complex block" do
+      input = """
+      defstyle "color-" <> color_name do
+        foo(true)
+        color(color_name)
```

Comment:
Is this the only form in which a variable can appear (`attr(<variable_name>)`)? 

If not, please clarify and provide more tests.

Reply:
In this instance `color_name` is a reference to the `color_name` pattern matched in the class name above. This is why we need the AST value so that I can programmatically build the function definitions.

Reply:
I'll take that as a yes, thanks

Reply:
Almost answers the question, I'm trying to understand if variables will only ever show up in the form `attr(<variable_name>)` eg `color(color_name)`. Does the parser need to worry about nesting `attr(attr(<variable_name>))` or attributes with multiple arguments `attr(arg1, <variable_name>)`?

Or put differently, if `color_name` is the variable used in the pattern match, should every single occurrence of `color_name` be replaced by the AST? What should the following parse to:

```elixir
"color-" <> color do
  foo(true)
  color(color)
  bar(false)
end
```

Is there a list of invalid variable names?

Reply:
So in specific case of `attr(<value>)` it shouldn't be nested such as `attr(attr(color_name))` the reason is because `attr/1` is a special call that will provide a tuple of `["Attr", value]` that the client will use to reference an attribute value on the original element. So `attr(attr(color_name))` doesn't make sense and we shouldn't parse that.

However, nested function calls are permitted. I don't have specific example but SwiftUI modifiers permit nested function calls and we are testing for that here: https://github.com/TwistingTwists/swift_class/blob/master/test/swift_class_test.exs#L114-L119

Reply:
Oops, I'm using `attr` to refer to any attribute not specifically the one that turns into `["Attr", value]`. 

How's this:

I'm trying to understand if variables will only ever show up in the form `<style_name>(<variable_name>)` eg `color(color_name)`. Does the parser need to worry about nesting `<style1_name>(<style2_name>(<variable_name>)) or attributes with multiple arguments `<style_name>(<arg1>, <variable_name>)`?

Reply:
Is it valid to have a variable name that matches an existing style name? For example is the following valid:
```
```

Reply:
Variable can show up anywhere in the arg list. This is why I changed the render of string literals from `red` to `"red"`.

Yes, multiple args are supported as this is represented in the tests: https://github.com/TwistingTwists/swift_class/blob/master/test/swift_class_test.exs#L33

Reply:
@NduatiK for:

```elixir
"color-" <> color do
color(color)
end
```

yes this would be valid

Reply:
Understood, I'll make the change.

Reply:
> Variable can show up anywhere in the arg list. This is why I changed the render of string literals from `red` to `"red"`.

@bcardarella, this was a really useful change, I've added a [`variable` combinator](https://github.com/TwistingTwists/swift_class/pull/6/files#diff-989ae876e444972b30d22012f2a3b4bfc233423d230ba87dc0b4ef6f53dd6232R29-R33) that matches an unquoted strings and wraps it in an AST.

---

I think this should be okay to merge now.

Reply:
😄 Np